### PR TITLE
Effective Date and Time for Unit Notes

### PR DESCRIPTION
**Pull request by dimak1**
*Issue #:* 
- bcgov/entity#16513

*Description of changes:*
- Add Effective Date component
- Set effective and expiry dates
- Add unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1415-gura6rd7.web.app

File: ppr-ui/tests/unit/EffectiveDateTime.spec.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+function createComponent (propsData: any): Wrapper<any> {
+  const localVue = createLocalVue()
+  localVue.use(Vuetify)
+
+  return mount((EffectiveDateTime as any), {
+    localVue,
+    propsData,
+    vuetify
+  })
+}
```
Comment (cameron-eyds) [inline ppr-ui/tests/unit/EffectiveDateTime.spec.ts]: I believe there is a common function from the test utils folder we can use. 
Trying to get away from having this function in every test file.

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue [hunk 2]
```diff
@@ -147,10 +154,21 @@ export default defineComponent({
       setMhrUnitNoteRegistration({ key: 'submittingParty', value: val })
     }
 
+    const handleEffectiveDateUpdate = (val: string) => {
+      setMhrUnitNote({ key: 'effectiveDateTime', value: val })
+      // expiry date is 90 days from effective date
+      const expiryDateTime = new Date(new Date(val).getTime() + 90 * 24 * 60 * 60 * 1000).toISOString()
+      setMhrUnitNote({ key: 'expiryDateTime', value: expiryDateTime })
```
Comment (cameron-eyds) [inline ppr-ui/src/components/unitNotes/UnitNoteReview.vue:161]: The UI creates/sets the expiry date? I assumed it just passed the effectiveDateTime to the api 🤔

File: ppr-ui/src/components/unitNotes/EffectiveDateTime.vue [hunk 3]
```diff
@@ -0,0 +1,236 @@
+<template>
+  <div id="contact-info-container">
```
Comment (cameron-eyds) [inline ppr-ui/src/components/unitNotes/EffectiveDateTime.vue]: copy pasta carry over i think

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue (hunk 2 above)
Comment (doug-lovett) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue:161]: CAUC can submit an expiry date, CAUE must submit an expiry date.

**Review by doug-lovett — APPROVED** [review]
LGTM

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue:161]: Is there implications if the user modifies there local system time here?

File: ppr-ui/tests/unit/EffectiveDateTime.spec.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    const wrapper: Wrapper<any> = createComponent(props)
+
+    const EffectiveDateTimeComponent = wrapper.findComponent(EffectiveDateTime)
+
+    expect(EffectiveDateTimeComponent.exists()).toBeTruthy()
+    expect(EffectiveDateTimeComponent.findComponent(SharedDatePicker).exists()).toBeTruthy()
+    expect(EffectiveDateTimeComponent.find(getTestId('time-picker-fields')).exists()).toBeTruthy()
+    expect(EffectiveDateTimeComponent.find(getTestId('footnote-label')).exists()).toBeFalsy()
+  })
+})
```
Comment (cameron-eyds) [inline ppr-ui/tests/unit/EffectiveDateTime.spec.ts:68]: Is it possible to add another test or two, to prove the time selected and/or format?

**Review by cameron-eyds — APPROVED** [review]
Looks good, just a few comments/questions

**Review by orelbn — APPROVED** [review]
Amazing job looks awesome!

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue (hunk 2 above)
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue:161]: @cameron-eyds I thought so too, but then I looked at API spec again, and there was the expiry date that we need to submit, so I added it here. There may be implications if local time on machine changes.
Maybe we can pass a flag (and not the date/time string) that the effective date is immediate? Then based on that server generates effective and exp date? For past date, we can just pass a date/string in past. The only problem I see is that the user will not see the (server time) summary note about the Notice at the bottom of the component.

File: ppr-ui/tests/unit/EffectiveDateTime.spec.ts (hunk 4 above)
Comment (dimak1) [reply ppr-ui/tests/unit/EffectiveDateTime.spec.ts:68]: Added more tests.

File: ppr-ui/src/components/unitNotes/EffectiveDateTime.vue (hunk 3 above)
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/EffectiveDateTime.vue]: Updated

File: ppr-ui/tests/unit/EffectiveDateTime.spec.ts (hunk 1 above)
Comment (dimak1) [reply ppr-ui/tests/unit/EffectiveDateTime.spec.ts]: Updated.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1415-gura6rd7.web.app

**Comment by dimak1** [comment]
Pending UXA feedback. Will merge after.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1415-gura6rd7.web.app

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1415-gura6rd7.web.app

**Comment by dimak1** [comment]
Pending another round of UXA.

**Pull request merged**